5_evaluate/src/eval_utility_fxns.R

```diff
@@ -115,62 +121,68 @@ prep_data_for_eval <- function(pred_obs, surface_cutoff_depth, middle_cutoff_dep
               n_dates = n(),
-              n_sites = length(unique(site_id)))
+              n_sites = length(unique(site_id))) %>%
+    mutate(depth_class = depth_class, .before=1)
 }
 
 #' @title Calculate rmse
 #' @description Calculate the rmse of model predictions over a 
 #' specified `grouping_var`
 #' @param eval_pred_obs a tibble of the matched model predictions and 
 #' observations, along with grouping variables for evaluation
-#' @param grouping_var the variable by which to group `pred-obs`
+#' @param grouping_var the variable by which to group `eval_pred_obs`
 #' before calculating the rmse
+#' @param depth_class the depth bin for the matched `eval_pred_obs`
 #' @return a tibble grouped by the grouping_var, with a column
 #' for rmse
-calc_rmse <- function(eval_pred_obs, grouping_var) {
+calc_rmse <- function(eval_pred_obs, grouping_var, depth_class) {
   eval_pred_obs %>%
     group_by(!!sym(grouping_var)) %>%
     summarize(rmse = sqrt(mean((pred_diff)^2, na.rm=TRUE)),
               n_dates = n(),
-              n_sites = length(unique(site_id)))
+              n_sites = length(unique(site_id))) %>%
+    mutate(depth_class = depth_class, .before=1)
 }
 
 #' @title Plot evaluation metrics as a bar plot
 #' @description generate a barplot of an evaluation metric, using the
 #' specified x and y variables
 #' @param plot_df a tibble of the matched model predictions and observations,
 #' along with grouping variables for evaluation
+#' @param num_eval_sites The number of unique evaluation sites 
 #' @param driver the name of the driver used to generate the
 #' model predictions
 #' @param y_var the variable for the y-axis of the plot
 #' @param y_label the label for the y-axis of the plot
 #' @param x_var the variable for the x-axis of the plot
-#' @param depth_class depth_class of the pred-obs (surface, middle,
-#' or bottom) that are being plotted.
+#' @faceting_variable variable to use for faceting the plot
 #' @param outfile The filepath for the exported png
 #' @return The filepath of the exported png 
-plot_evaluation_barplot <- function(plot_df, driver, y_var, y_label, x_var, depth_class, outfile) {
+plot_evaluation_barplot <- function(plot_df, num_eval_sites, driver, y_var, y_label, x_var, faceting_variable, outfile) {
   bar_plot <- plot_df %>%
     ggplot(aes(x = get(x_var), y = get(y_var))) +
     geom_col(fill='cadetblue3', color='cadetblue4') +
-    labs(title= sprintf("%s %s predictions: %s by %s", driver, depth_class, y_var, x_var), 
+    labs(title = paste(sprintf("%s predictions: %s by %s", driver, y_var, x_var),
+                       sprintf("Total # of evaluation sites: %s", num_eval_sites), sep ='\n'), 
          x=sprintf("%s", x_var), 
          y=sprintf("%s (\u00b0C)", y_label)) +
+    facet_wrap(~get(faceting_variable), nrow = 2) +
     theme_bw()
   
-  ggsave(filename=outfile, plot=bar_plot, dpi=300, width=10, height=6)
+  ggsave(filename=outfile, plot=bar_plot, dpi=300, width=10, height=8)
```

Comment:
Just a thought, but for figure exports, I like to put args like this in the function (e.g., plot_evaluation_barplot(..., heigh = 8)) with defaults because we do tend to mess with these. If they have defaults, you don't _have_ to specify a value when defining the target, but if you'd like to, you can (even as a more "global" variable at the top of `_targets.R` if you want to change all of them at once). Probably just a style preference thing. The only downside I see with the current approach is that you can't control the individual plots if you'd like to (like having DoY be wider than time, etc).

Reply:
Oh that does offer some nice flexibility. I added them as default arguments.